### PR TITLE
[task] treat cwd consistently as a path

### LINKED ISSUE #3883: [task] suspicious handling of cwd on windows
It used to work on until two weeks ago but stopped working a few days ago. I have no idea why. Maybe that's a change in the underlying VSCode class Uri. Here's the issue: 

In the task runner [we convert the cwd from the taskConfig to a path](https://github.com/theia-ide/theia/blob/0ce38188ed3010662b90feacd3097d1b7eeb36c1/packages/task/src/node/process/process-task-runner.ts#L83), then we hand it into findCommand and treat it as a URI ([1](https://github.com/theia-ide/theia/blob/0ce38188ed3010662b90feacd3097d1b7eeb36c1/packages/task/src/node/process/process-task-runner.ts#L151), [2](https://github.com/theia-ide/theia/blob/0ce38188ed3010662b90feacd3097d1b7eeb36c1/packages/task/src/node/process/process-task-runner.ts#L163)). This fails on windows as `C:\badabada` is not a valid URI.

### PR DESCRIPTION
**Pull request by JanKoehnlein**
Fixes #3883

### CODE REVIEW

**Comment by simark** [comment]
Could you also add a test?  IIRC we have those tests in task-server.slow-spec.ts.

**Comment by JanKoehnlein** [comment]
I've fixed the path conversion in the test (which converted windows paths in a wrong way).
For backward compatibility, I've also allowsed passing a file:/ URI instead of an fsPath.

File: packages/task/src/node/task-server.slow-spec.ts [hunk 1]
```diff
@@ -125,6 +124,17 @@ describe('Task server / back-end', function () {
         await p;
     });
 
+    it('task is executed successfully with cwd as a file URI', async function () {
+        const command = isWindows ? commandShortrunningindows : commandShortRunning;
+        const config = createProcessTaskConfig('shell', FileUri.fsPath(command), []);
+        (config as any).cwd = FileUri.create(wsRoot).toString();
```
Comment (simark) [inline packages/task/src/node/task-server.slow-spec.ts]: This doesn't build for me:

    lerna ERR! [lint] ERROR: /home/emaisin/src/theia/packages/task/src/node/task-server.slow-spec.ts:130:20 - Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.


I suggest adding an optional `cwd` parameter to `createProcessTaskConfig`.

Comment (simark) [inline packages/task/src/node/task-server.slow-spec.ts]: This doesn't build for me:

    lerna ERR! [lint] ERROR: /home/emaisin/src/theia/packages/task/src/node/task-server.slow-spec.ts:130:20 - Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.


I suggest adding an optional `cwd` parameter to `createProcessTaskConfig`.

**Comment by JanKoehnlein** [comment]
@simark In fact, the cwd is now tested by all task test that run the test scripts, as they are found via cwd. 

Please note that the test was wrong before, as it treated windows file paths as URIs, i.e. `c:\some\path` became a URI with the scheme `c`, and that URI was converted back to a file system path by stripping the drive letter. On access, the process task runner assumed the default drive `c:`. So tests were green even though we had various bugs on the way. 

Feel free to add more test if you want to. But note that in order to properly test such things we'd have to at least make sure that our windows test machine allowed us to access a drive other than `c`. Given the state of our current appveyor tests, that's a bit out of reach.

**Comment by simark** [comment]
> @simark In fact, the cwd is now tested by all task test that run the test scripts, as they are found via cwd.
> 
> Please note that the test was wrong before, as it treated windows file paths as URIs, i.e. `c:\some\path` became a URI with the scheme `c`, and that URI was converted back to a file system path by stripping the drive letter. On access, the process task runner assumed the default drive `c:`. So tests were green even though we had various bugs on the way.

Ok, thanks for the precision!

**Pull request merged**